portal/portal-charon/charon/src/webapp/scripts/themeswitcher.js

```diff
@@ -1,43 +1,97 @@
-function themeSwitcher(){
+function sakaiThemeSwitcher(){
 
-    let themeSwitcher = document.getElementById("themeSwitcher");
+    let darkThemeSwitcher = document.getElementById("sakai-darkThemeSwitcher");
 
-    SAK_THEME_SELECTION_KEY = portal.user.id + "#theme";
+    const defaultThemeClass = 'sakaiUserTheme-notSet';
+    const lightThemeClass = 'sakaiUserTheme-light';
+    const darkThemeClass = 'sakaiUserTheme-dark';
 
-    if(themeSwitcher != null) {
+    init();
+    
+    function init() {
+        // attach listener to dark theme toggle switch
+        darkThemeSwitcher && darkThemeSwitcher.addEventListener('click', toggleDarkTheme, false);
+
+        if(isLoggedIn()) {
+            // only check for unset theme preference because light and dark themes are already set by Java
+            if(isPortalThemeUserPrefUnset()) {
+                if(isOsDarkThemeSet()) {
+                    enableDarkTheme();
+                } else {
+                    // to define a user preference:
```

Comment:
PEDANT ALERT: I like a single space after the if. You've done that in some places.

Reply:
Good catch. Thanks.